### PR TITLE
Mongo Searchable Prefixes via BiConsumer

### PR DESCRIPTION
**Pull request by idlira**
Offers the ability to create Transformers to offer an alternative way to collect values to index from Properties, where the standard `getValue()` is not applicable.

Also adds the _Optimus Prime_ , _Megatron_ and _Bumblebee_ transformers for MultiLanguageString, StringList and StringMap respectively 😉

### CODE REVIEW

File: src/main/java/sirius/biz/mongo/MultiLanguageStringPropertyTransformer.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    public Class<PrefixSearchableContentSupplier> getTargetClass() {
+        return PrefixSearchableContentSupplier.class;
+    }
+
+    @Nullable
+    @Override
+    @SuppressWarnings("unchecked")
+    public PrefixSearchableContentSupplier make(@Nonnull MultiLanguageStringProperty source) {
+        return entity -> {
+            StringList tokens = new StringList();
```
Comment (andyHa) [inline src/main/java/sirius/biz/mongo/MultiLanguageStringPropertyTransformer.java]: why this intermediate step, couldn't we transform into a BiConsumer which accepts an entity and a StringBuilder or even BiConsumer<PrefixSearchableEntity, Consumer<String>> and then invoke with (entity, this::addContentString) or the like

Comment (andyHa) [reply src/main/java/sirius/biz/mongo/MultiLanguageStringPropertyTransformer.java]: or simply transform to String

Comment (idlira) [reply src/main/java/sirius/biz/mongo/MultiLanguageStringPropertyTransformer.java]: We index not only the words, but the phrase as well (see test case)
Indigo blue becomes: ["indigo blue", "indigo", "blue"]

For Multi-Language scenario we want each translation "phrase" to be processed independently and not only its loose words...

Comment (idlira) [reply src/main/java/sirius/biz/mongo/MultiLanguageStringPropertyTransformer.java]: Implemented as `BiConsumer`. Hopefully nailed it now 😬

File: src/main/java/sirius/biz/mongo/PrefixSearchableContentConsumer.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Used as target class to transform {@link sirius.db.mixing.Property properties} into a {@link BiConsumer}.
+ * <p>
+ * The consumer takes the {@link PrefixSearchableEntity} containing the values to extract which are then
+ * passed as the {@link Consumer} argument.
+ */
+public interface PrefixSearchableContentConsumer extends BiConsumer<PrefixSearchableEntity, Consumer<Object>> {
```
Comment (andyHa) [inline src/main/java/sirius/biz/mongo/PrefixSearchableContentConsumer.java]:
```suggestion
public interface PrefixSearchableContentConsumer extends BiConsumer<PrefixSearchableEntity, Consumer<String>> {
```

Comment (idlira) [reply src/main/java/sirius/biz/mongo/PrefixSearchableContentConsumer.java]: +1

**Pull request merged**